### PR TITLE
Refactor `mlos_bench.storage` and add `TunableConfigTrialGroup` property for `TrialData` and `ExperimentData`

### PR DESCRIPTION
**Pull request by bpkroth**
Useful for grouping trials by the config they used.  In use by upcoming #633 for generating graphs with variance error bars for repeated configs.

Also refactors a number of other things:
- Standardize on `experiment_id` instead of `exp_id`
  (but not the db schema for now)
- Standardize on `tunable_config_id` instead of `config_id` in the API since we also call it `tunable_config` for the object fetching property to distinguish from the `config` dict used internally.
  (but not the db schema for now)
- Rework the idea of `TunableConfigTrialGroup` as an object inaddition to an ID (further methods can be added later to move back and forth between types when doing interactive analysis).
- Rework the idea of a `TunableConfig` as an object for fetching tunable value assignments (similar justification - easier grouping in the future by fetching trial across experiments based on config - eventually could be used to house the experiment merge logic).
- Rename `results` APIs to `results_df` (similar for others that return `pandas.DataFrame`) to match the `results_dict` that return `dict`
- Refactor test fixtures to match other styles and for future use (moved to #644).
- Expand tests

NOTE:
- We cut a new version with this commit since there are potentially breaking API changes (e.g., `results` -> to `results_df` and `exp_id` -> `experiment_id`).

Currently builds off of #644 and splits work out of #633

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/storage/tunable_config_trial_group_data_test.py
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/storage/tunable_config_trial_group_data_test.py:1]: Need tests for `trials` and `results` properties

File: mlos_bench/mlos_bench/tests/storage/tunable_config_data_test.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#
+"""
+Unit tests for loading the TunableConfigData.
+"""
+
+from mlos_bench.storage.base_experiment_data import ExperimentData
+from mlos_bench.tunables.tunable_groups import TunableGroups
+
+
+def test_tunable_config_data(exp_data: ExperimentData,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/storage/tunable_config_data_test.py:13]: Rename: `test_trial_data_tunable_config_data`

File: mlos_bench/mlos_bench/tests/storage/fixtures.py
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/storage/fixtures.py:1]: Mistakenly kept in the last PR - was moved to `storage/sql/`.

File: mlos_bench/mlos_bench/tests/storage/exp_data_test.py [hunk 2]
```diff
@@ -62,3 +63,28 @@ def test_exp_trial_data_objectives(storage: Storage,
         "opt_target": "some-other-target",
         "opt_direction": "max",
     }
+
+
+def test_exp_data_tunable_config_trial_group_id(exp_data: ExperimentData) -> None:
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/storage/exp_data_test.py:68]: rename - tests `results_df` property

File: mlos_bench/mlos_bench/storage/sql/tunable_config_trial_group_data.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        return self._experiment_id
+
+    @property
+    def tunable_config_id(self) -> int:
+        """
+        ID of the config.
+        """
+        return self._tunable_config_id
+
+    def _get_tunable_config_trial_group_id(self) -> int:
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/storage/sql/tunable_config_trial_group_data.py:61]: test this vs. bulk load version

**Review by motus — APPROVED** [review]
LGTM!

**Pull request merged**